hera_sim/sigchain.py

```diff
@@ -1150,30 +1140,16 @@ def build_coupling_matrix(
                     axis2_inds=horizon_select, inplace=False, run_check=False
                 )
 
-        # Now make sure we're OK to interpolate the beam. AnalyticBeam makes
-        # this a little annoying... but whatever.
-        if hasattr(uvbeam, "interpolation_function"):
-            if uvbeam.interpolation_function is None:
-                uvbeam.interpolation_function = pixel_interp
-        else:
-            uvbeam.interpolation_function = pixel_interp
-        if hasattr(uvbeam, "freq_interp_kind"):
-            if uvbeam.freq_interp_kind is None:
-                uvbeam.freq_interp_kind = freq_interp
-        else:
-            uvbeam.freq_interp_kind = freq_interp
-
         # Now we'll actually interpolate the beam.
         # The end shape is (n_az, n_freq, 2, 2).
-        jones_matrices = (
-            uvbeam.interp(
-                az_array=unique_angles,
-                za_array=np.ones_like(unique_angles) * np.pi / 2,
-                freq_array=freqs * units.GHz.to("Hz"),
-            )[0]
-            .squeeze()
-            .transpose(3, 2, 1, 0)
-        )
+        uvbeam = BeamInterface(uvbeam)
```

Comment:
Is this really a UVBeam or can it also be an AnalyticBeam? If it can be either this name is very confusing.

Reply:
It's true that it can be an analytic beam, but since the name is API-facing, I don't want to change it here.